### PR TITLE
added evaluator and evaluatee

### PR DESCRIPTION
**Pull request by Zt0**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,19 +9,27 @@
       "schema:sync": "ts-node ./node_modules/typeorm/cli.js schema:sync",
       "start": "nodemon src/index.ts",
       "typeorm": "node --require ts-node/register ./node_modules/typeorm/cli.js",
-      "prettier-format": "prettier --config .prettierrc 'src/**/*.ts' --write",
-      "eslint": "eslint src --ext .ts  --fix"
+      "fix": "npm run prettier -- --write && npm run eslint -- --fix",
+      "prettier": "prettier --config .prettierrc 'src/**/*.ts' --write",
```
Comment (harutyunyan611) [inline package.json]: Difference between this two?

File: package.json [hunk 2]
```diff
@@ -9,19 +9,27 @@
...
+      "lint:fix": "eslint --fix --ext .js,.ts, .",
+      "eslint": "eslint --ignore-pattern '.eslintrc.js' --ext .js,.ts ."
```
Comment (harutyunyan611) [inline package.json]: Difference between this two?

File: src/events/dto/elementIdDto.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import { IsInt } from 'class-validator';
-import { IIdRefDto } from '../interface/idRefInterface';
+import {IsInt} from 'class-validator'
+import {IElementIdDto} from '../interface/idRefInterface'
 
-export class elementIdDto implements IIdRefDto {
+export class elementIdDto implements IElementIdDto {
```
Comment (harutyunyan611) [inline src/events/dto/elementIdDto.ts:4]: What is element

**Review by harutyunyan611 — CHANGES_REQUESTED** [review]

**Review by harutyunyan611 — DISMISSED** [review]

File: src/events/controller/criteriaController.ts [hunk 4]
```diff
@@ -9,51 +9,56 @@ import {
...
   @Inject()
-  criteriaService: CriteriaService;
+  criteriaService: CriteriaService
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Post()
-  create(@Body() createCriteriaDto: CreateCriteriaDto): Promise<Criteria> {
-    return this.criteriaService.create(createCriteriaDto);
+  @Put(':id/subcriteria')
+  addSubCriteria(
```
Comment (ArthurO2002) [inline src/events/controller/criteriaController.ts]: change :id to eventId to be more clear

File: src/events/controller/criteriaController.ts [hunk 5]
```diff
@@ -9,51 +9,56 @@ import {
...
-  ): Promise<Criteria> {
-    return this.criteriaService.update(id, updateCriteriaDto);
+  @Post()
+  create(@Body() createCriteriaDto: CreateCriteriaDto): Promise<Criteria> {
+    return this.criteriaService.create(createCriteriaDto)
+  }
+
+  @UseInterceptors(ClassSerializerInterceptor)
+  @Put(':id')
+  update(@Param('id') id: number, @Body() updateCriteriaDto: UpdateCriteriaDto): Promise<Criteria> {
```
Comment (ArthurO2002) [inline src/events/controller/criteriaController.ts:55]: use patch

File: src/events/controller/eventController.ts [hunk 6]
```diff
@@ -8,130 +8,104 @@ import {
...
   }
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Get()
-  async findAll(): Promise<EventPivotDto[]> {
-    return (await this.eventsService.findAll()).map((event) =>
-      eventGetDto(event)
-    );
+  @Put(':id/rating')
+  addRating(@Param('id') eventId: number, @Body() ratingRef: elementIdDto): Promise<Event> {
```
Comment (ArthurO2002) [inline src/events/controller/eventController.ts]: use patch here too

File: src/events/controller/eventController.ts [hunk 7]
```diff
@@ -8,130 +8,104 @@ import {
...
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Get('/allByTitle/:title')
-  async findAllByTitle(
-    @Param('title') title: string
-  ): Promise<EventPivotDto[]> {
-    return (await this.eventsService.findAllByTitle(title)).map((event) =>
-      eventGetDto(event)
-    );
+  @Put(':id/criteria')
```
Comment (ArthurO2002) [inline src/events/controller/eventController.ts:73]: .

File: src/events/controller/eventController.ts [hunk 8]
```diff
@@ -8,130 +8,104 @@ import {
...
+  addCriteria(@Param('id') eventId: number, @Body() criteriaRef: elementIdDto): Promise<Event> {
+    return this.eventsService.addCriteria(eventId, criteriaRef)
   }
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Get(':id')
-  async findOneById(@Param('id') id: number): Promise<EventPivotDto> {
-    return eventGetDto(await this.eventsService.findOneById(id));
+  @Put(':id/subCriteria')
```
Comment (ArthurO2002) [inline src/events/controller/eventController.ts:79]: .

File: src/events/controller/eventController.ts [hunk 9]
```diff
@@ -8,130 +8,104 @@ import {
...
+    @Body() criteriaRef: ISubCriteriaRef,
+  ): Promise<void> {
+    return this.eventsService.addSubCriteria(eventId, criteriaRef)
   }
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Get('/byTitle/:title')
-  async findOneByTitle(@Param('title') title: string): Promise<EventPivotDto> {
-    return eventGetDto(await this.eventsService.findOneByTitle(title));
+  @Put(':id/evaluator')
```
Comment (ArthurO2002) [inline src/events/controller/eventController.ts:88]: .

File: src/events/controller/eventController.ts [hunk 10]
```diff
@@ -8,130 +8,104 @@ import {
...
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Get('/byTimePeriod/:timePeriod')
-  async findOneByTimePeriod(
-    @Param('TimePeriod') timePeriod: Period
-  ): Promise<EventPivotDto> {
-    return eventGetDto(
-      await this.eventsService.findOneByTimePeriod(timePeriod)
-    );
+  @Put(':id/evaluatee')
```
Comment (ArthurO2002) [inline src/events/controller/eventController.ts:94]: .

File: src/events/dto/EventEvaluateeGetDto.ts [hunk 11]
```diff
@@ -0,0 +1,8 @@
+import {eventGetDto} from './eventGetDto'
+import {userGetDto} from '../../users/dto/userGetDto'
+import {EventEvaluatee, EventEvaluateeGetDto} from '../entity/eventEvaluatee'
+;``
```
Comment (ArthurO2002) [inline src/events/dto/EventEvaluateeGetDto.ts:4]: hn?

File: src/events/dto/eventCreateDto.ts [hunk 12]
```diff
@@ -6,32 +6,32 @@ import {
...
 export class CreateEventDto {
   @IsString()
-  readonly title: string;
+  readonly title: string
 
   @IsInt()
   @Min(0)
   @Max(100)
-  readonly bonus: number;
+  readonly bonus: number
```
Comment (ArthurO2002) [inline src/events/dto/eventCreateDto.ts:23]: Are you sure this fields are readonly?

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

**Review by harutyunyan611 — APPROVED** [review]

**Pull request merged**